### DISCUSSION TITLE
Docs: Discourage the use of "local images"

### DISCUSSION THREAD

**Discussion by FleetAdmiralJakob** [vercel/next.js#72637] [App Router] [answered]
### What is the documentation issue?

People often use so-called "local images", "relative image imports", or "statically imported images" (the naming is a thing to talk about another time) to import their image into the `<Image />` component which itself even accepts the `StaticImageData` type. This often leads to issues when switching to Turbopack like this error:

```
Module not found: Can't resolve 'whatever'
server relative imports are not implemented yet. Please try an import relative to the file you are importing from.
```

As there is no fix planned for this in Turbopack as seen in these issues: https://github.com/vercel/turborepo/issues/4979, https://github.com/vercel/turborepo/issues/3573, https://github.com/vercel/next.js/issues/46358, https://github.com/vercel/next.js/issues/47919

I would discourage people from using this, adding a hint or note to the docs and only keeping it for backwards compatibility.

### Is there any context that might help us understand?

It seems to never should have been a feature as shown in this comment:

> When we discussed this with the Next.js team, they expressed that this was an unintentional "feature" and isn't actually supposed to work by default. Instead, they recommend doing one of the above suggestions. The benefit of this is that if follows the conventions in the larger ecosystem, which is quickly adopting TS's paths, package.json's imports field, and import maps in general.

https://github.com/vercel/turborepo/issues/4979#issuecomment-1551898170

### Related docs pages
https://nextjs.org/docs/app/building-your-application/optimizing/images#local-images
https://nextjs.org/docs/app/api-reference/components/image#src

**Comment by samcx** [comment, marked as answer]
@FleetAdmiralJakob It's unclear what the issue is exactly here—if you're trying to use an image from '/public', I believe the image `src` has always been from the root (e.g., your src is `/image.jpeg` if the image lives in `/public/image.jpeg`).

Usage of local images works the same in Turbopack:

```
import retro from './retro.svg'
```

![CleanShot 2024-11-12 at 14 55 12@2x](https://github.com/user-attachments/assets/3e96f6f9-2256-4300-8692-e632d8d194ae)

**Reply by FleetAdmiralJakob** [reply]
lol, I think I was just stupid. Maybe not. But I think you are correct. Sorry
